src/rgw/rgw_tracer.cc

```diff
@@ -0,0 +1,117 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+#include "rgw_tracer.h"
+#include <yaml-cpp/yaml.h>
+
+namespace tracing {
+
+  const std::shared_ptr<opentracing::Tracer> Tracer::noop_tracer = opentracing::MakeNoopTracer();
+  TracerConfiguration tracing::Tracer::tracer_conf;
+
+  Tracer::Tracer(jaegertracing::Config& conf):open_tracer(jaegertracing::Tracer::make(conf)) {};
+
+  Tracer::Tracer(const std::string& config_file_path) {
+    auto config_yaml = YAML::LoadFile(config_file_path);
+    auto config = jaegertracing::Config::parse(config_yaml);
+    open_tracer =  jaegertracing::Tracer::make(config);
+  };
+
+  Tracer::~Tracer() {open_tracer->Close();}
+
+  std::unique_ptr<Span> Tracer::start_trace(const std::string& trace_name) {
+    std::unique_ptr<opentracing::Span> span;
+    if(is_enabled()) {
+      span = open_tracer->StartSpan(trace_name);
+    }
+    else {
+      span = noop_tracer->StartSpan(trace_name);
+    }
+    return std::unique_ptr<Span>(new Span(span));
+  }
+
+  std::unique_ptr<Span> Tracer::start_span(const std::string& span_name, std::unique_ptr<Span>& parent_span) {
+    std::unique_ptr<opentracing::Span> span;
+    if(is_enabled()) {
+      span = open_tracer->StartSpan(span_name, {parent_span->get_reference()});
+    }
+    else {
+      span = noop_tracer->StartSpan(span_name);
+    }
+    return std::unique_ptr<Span>(new Span(span));
+  }
+
+  bool Tracer::is_enabled() const {
+    return tracer_conf.is_enabled();
+  }
+
+  Span::Span(std::unique_ptr<opentracing::Span>& _span):span(_span.release()) {};
+
+  Span::~Span() {span->Finish();}
```

Comment:
@zenomri is this going to send the span to the agent every time we hit the dtor?
how is it done with the global tracer from opentracing?
do they send on every span or only when the trace ends for all child spans?

Reply:
I took a look at how it happens.
The `jaegertracing::Span` d'tor is already implemented to finish spans.  so the wrapping doesn't really help us.  we can dump it. 
the global tracer of opentracing doesn't take care of sending the spans. 
once the `Finish` method of a span is called, the tracer uses some reporter object that sends the span.

Reply:
ok. so lets remove this wrapper

Reply:
removed in ffd1262